bson/src/main/org/bson/internal/ProvidersCodecRegistry.java

```diff
@@ -39,11 +41,16 @@ public ProvidersCodecRegistry(final List<? extends CodecProvider> codecProviders
 
     @Override
     public <T> Codec<T> get(final Class<T> clazz) {
-        return get(new ChildCodecRegistry<T>(this, clazz));
+        return get(new ChildCodecRegistry<T>(this, clazz, null));
     }
 
     @Override
     public <T> Codec<T> get(final Class<T> clazz, final List<Type> typeArguments) {
+        notNull("typeArguments", typeArguments);
+        isTrueArgument("typeArguments is not empty", typeArguments.size() >= 1);
```

Comment:
`typeArguments.isEmpty()` is both more explicit than `typeArguments.size() >= 1` and for some implementations may be more performant (a silly example: if `LinkedList` were not to cache its size, then `LinkedList.size` would have had linear time complexity). While the latter is irrelevant here, readability is a good enough argument.